### PR TITLE
Documentation Comments 

### PR DESCRIPTION
**Pull request by cheeyixu**
Please disregard v2 as I made a mistake there. I am still figuring out how to delete a commit. The final version is v2.1

v2.1.1: fixed escape characters in xml. trimmed trailing spaces and formatted the doc such that in between each methods there are 2-line spaces

### CODE REVIEW

File: BigInteger/BigInteger.cs [hunk 1]
```diff
@@ -282,11 +284,11 @@ public BigInteger(System.Collections.Generic.IList<byte> inData, int length = -1
         while (dataLength > 1 && data[dataLength - 1] == 0)
             dataLength--;
     }
-
-
-    //***********************************************************************
-    // Constructor (Default value provided by an array of unsigned integers)
-    //*********************************************************************
+
```
Comment (bazzilic) [inline BigInteger/BigInteger.cs:288]: Check the file for trailing spaces please

File: BigInteger/BigInteger.cs [hunk 2]
```diff
@@ -601,9 +623,13 @@ public BigInteger(uint[] inData)
 
 
 
-    //***********************************************************************
-    // Overloading of unary << operators
-    //***********************************************************************
+
+    /// <summary>
+    /// Overloading of the unary &lt&lt operator, support BigInteger to have left-shifted
```
Comment (bazzilic) [inline BigInteger/BigInteger.cs]: I think there should be `;` symbols after `&lt`, `&gt`, etc. Please check.

File: BigInteger/BigInteger.cs [hunk 3]
```diff
@@ -1128,9 +1196,13 @@ public override int GetHashCode()
...
-    //***********************************************************************
-    // Overloading of bitwise AND operator
-    //***********************************************************************
+
+    /// <summary>
+    /// Overloading of bitwise AND operator, support bitwise and operation on BigInteger type
+    /// </summary>
+    /// <param name="bi1">First BigInteger</param>
+    /// <param name="bi2">Second BigInteger</param>
+    /// <returns>BigInteger result after performing & operation</returns>
```
Comment (bazzilic) [inline BigInteger/BigInteger.cs]: Not sure, please check if you need to escape the `&` symbol in docs

File: BigIntegerTests/Operators.cs [hunk 4]
```diff
@@ -456,6 +456,7 @@ public void TestDecrement()
         [TestMethod]
         public void TestShiftRight()
         {
+            (new BigInteger()).ToHexString();
```
Comment (bazzilic) [inline BigIntegerTests/Operators.cs]: What's this?

Comment (cheeyixu) [reply BigIntegerTests/Operators.cs]: Sorry about that. I wanted to test the method but have forgotten to delete the line.

**Pull request merged**